MyTasks/Task2_1/UserAndBotDialogue.cs

```diff
@@ -0,0 +1,60 @@
+﻿
+Console.WriteLine("Здравствуйте! Введите Ваше имя:");
+var userName = Console.ReadLine();
+
+while (char.IsLower(userName, 0))
+{
+    Console.WriteLine("Пожалуйста, введите имя с заглавной буквы:");    // "обработка" неправильного пользовательского ввода
+    userName = Console.ReadLine();
+}
+
+foreach (var symbol in userName)   // где-то в этом блоке баг: если первый ввод некорректный, то идет
+                                    // бесконечный вывод "Имя должно состоять из букв" в ответ
+                                    // как на последующий некорректный ввод, так и на корректный.
+{
+    while (char.IsNumber(symbol) || char.IsPunctuation(symbol) || char.IsWhiteSpace(symbol))
+    {
+        Console.WriteLine("Имя должно состоять из букв.");
+        userName = Console.ReadLine();
+    }
+}
+
+Console.WriteLine("Введите Вашу фамилию:");
+var userSurname = Console.ReadLine();
+
+while (char.IsLower(userSurname, 0))
+{
+    Console.WriteLine("Пожалуйста, введите фамилию с заглавной буквы:");    // "обработка" неправильного пользовательского ввода
+    userSurname = Console.ReadLine();
+}
+
+foreach (var symbol in userSurname)    //где-то в этом блоке баг: смотри строки 10-12
+{
+    while (char.IsNumber(symbol) || char.IsPunctuation(symbol) || char.IsWhiteSpace(symbol))
+    {
+        Console.WriteLine("Фамилия должна состоять из букв.");
+        userSurname = Console.ReadLine();
+    }
+}
```

Comment:
You are getting each symbol from userName in foreach, then reading again in while loop if it doesnt fit. 
It's very hard logic for this case. Try to implement another logic. You can do the easiest solution
Here its not a bug, it is your self-made mistake. For example, in userSurname - "asd123" string. From foreach we are getting 1st symbol -> this symbol 'a' getting into while loop -> we are executing userSurname = Console.ReadLine();
But this call will not override userSurname(in frames of local variable in while loop) and will go again in while loop with same symbol 'a'. Because you started your foreach loop with userSurname = "asd123";

Reply:
Done.